### ISSUE TITLE
[Rule Request]: Private @IBAction opt-in rule

### ISSUE THREAD

**Issue by ornithocoder** [realm/SwiftLint#1931]
This is a request to introduce a new opt-in rule which warns against public IBActions. 

Triggering example:

```swift
@IBAction func ↓fooButtonTapped(_ sender: UIButton) {
    // ...
}
```

Non-triggering example:

```swift
@IBAction private func fooSegmentedControlChanged(_ sender: UISegmentedControl) {
    // ...
}
```

```swift
private extension Toto {
    @IBAction func fooSegmentedControlChanged(_ sender: UISegmentedControl) {
        // ...
    }
}
```
Workaround via custom rule (snippet from `.swiftlint.yml`) while the rule isn't implemented/merged into master:

```yml
custom_rules:
  private_ibaction:
    name: "Private @IBAction"
    regex: "@IBAction\\s+func"
    message: "@IBAction should be private"
    severity: warning
```

**Comment by ornithocoder** [comment]
@marcelofabri @jpsim let me know if this makes sense to you. In case it does, I'll work on it.

I'm my project we have all our `@IBAction` private, and we test the code by by sending actions (`.sendActions(_:)`) to the controllers (touch up inside, value changed, etc...).

**Comment by marcelofabri** [comment]
I think this would be a nice rule 👍

**Comment by jpsim** [comment]
Sounds reasonable as opt-in.

**Comment by ornithocoder** [comment]
Super. I'll work on it tomorrow morning :-) Thanks guys!

**Comment by marcelofabri** [comment]
Implemented in #1935.
